generator.rb

```diff
@@ -0,0 +1,67 @@
+def is_vowel?(letter)
+  vowels = ["a", "e", "i", "o", "u"]
+  return vowels.include?(letter)
+end
+
+def is_valid(input)
+  result = true
+  if input.length < 2
+    puts "Your input should be at least two characters."
+    result = false
+  end
+
+  if input !~ /[aeiou]/
+    puts "Your input should contain at least one vowel."
+    result = false
+  end
+
+  return result
+end
+
+def get_input
+  input = gets.chomp.downcase
+  until is_valid(input)
```

Comment:
This is so readable! Making a smaller method to check if the input was valid in `is_valid` is really great. Nice!